leshan-server-cluster/src/main/java/org/eclipse/leshan/server/cluster/RedisSecurityStore.java

```diff
@@ -97,11 +97,16 @@ public SecurityInfo getByIdentity(String identity) {
     public SecurityInfo add(SecurityInfo info) throws NonUniqueSecurityInfoException {
         byte[] data = serialize(info);
         try (Jedis j = pool.getResource()) {
-            j.set((SEC_EP + info.getEndpoint()).getBytes(), data);
             if (info.getIdentity() != null) {
                 // populate the secondary index (security info by PSK id)
+                String oldEndpoint;
+                if ((oldEndpoint = j.hget(PSKID_SEC, info.getIdentity())) != null
```

Comment:
In the Leshan code base, we generally don't initialize variable in `if` condition.
Could you please, move the initialization out of the if ?
```java
String oldEndpoint =  j.hget(PSKID_SEC, info.getIdentity();
if (oldEndpoint != null ...
```